src/requirements/data/exams/ExamCredit.ts

```diff
@@ -60,16 +60,34 @@ function userDataToCourses(
   return courses;
 }
 
-export default function getCourseEquivalents(
+function getCourseEquivalentsFromOneMajor(
   college: string,
   major: string,
   userData: ExamsTaken
-): CourseTaken[] {
+): readonly CourseTaken[] {
   const APCourseEquivalents = userDataToCourses(college, major, userData, 'AP');
   const IBCourseEquivalents = userDataToCourses(college, major, userData, 'IB');
   return APCourseEquivalents.concat(IBCourseEquivalents);
 }
 
+export default function getCourseEquivalentsFromUserExams(user: AppUser): readonly CourseTaken[] {
+  const courses: CourseTaken[] = [];
+  const examCourseIDSet = new Set<number>();
+  const userExamData: ExamsTaken = { AP: [], IB: [] };
+  user.exam.forEach(exam => {
+    userExamData[exam.type].push(exam);
+  });
+  user.major.forEach(major =>
+    getCourseEquivalentsFromOneMajor(user.college, major, userExamData).forEach(course => {
+      if (!examCourseIDSet.has(course.courseId)) {
+        examCourseIDSet.add(course.courseId);
+        courses.push(course);
+      }
```

Comment:
Why is the double counting accounted for here? Doesn't the requirements algorithm take care of that?

Reply:
Make sure that the same equivalent course doesn't count twice for requirement like academic credits.

Reply:
Hmm, then what about equivalent courses being double counted with non-AP/IB courses? For example, I'm a CS major and I take CHEM 2090. I also take AP CHEM. Would that similarly inflate requirements like academic credits?

Reply:
It might, although I'm not sure about the full implication yet.
It appears to me that we can't simply drop the AP exam when we also have taken the same equivalent class, because you might use AP exam to fulfill one stuff and the actual class to fulfill some other stuff.
This is something need further discussion. However, I think it shouldn't affect this diff, since fulfilling academic credits is usually the least thing people worry about.